### ISSUE TITLE
K6 incorrectly expect a Location header in 30x responses

### ISSUE THREAD

**Issue by cboitel** [grafana/k6#2704]
### Brief summary

K6 rejects request returning 3xx status codes if they do not provide a Location HTTP header.

As per RFC 7231, Location header isn't required to be returned and a body may also be provided.

### k6 version

0.40

### OS

Ubuntu (focal)

### Steps to reproduce the problem

1. setup a server that responds a 3XX response without a Location header
2. Send a request to such server

### Expected behaviour

As defined in  [RFC7231](https://www.rfc-editor.org/rfc/rfc7231#section-6.4), response should return a location not MUST

1. if Location is provided, honor request's params.redirects
2. if no Location is provided, provide response status, headers, body and tls properties 

### Actual behaviour

1. response status is 0
2. response headers is {}
3. response body is null
4. response tls_properties are left empty
5. response error is "30x response missing Location header"

**Comment by atlekbai** [comment]
Hello!
I have tried to reproduce the error, but I couldn't.
I have setup a simple HTTP server with Go:

```go
package main

import (
	"fmt"
	"log"
	"net/http"
)

func main() {
	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		w.WriteHeader(http.StatusMovedPermanently)
		fmt.Fprintf(w, "Hello!")
	})

	fmt.Printf("Starting server at port 8181\n")
	if err := http.ListenAndServe(":8181", nil); err != nil {
		log.Fatal(err)
	}
}
```

My `script.js`:
```js
import http from "k6/http";

export default function() {
    let response = http.get("http://localhost:8181/");
    console.log(response.status)
    console.log(response.headers)
    console.log(response.body)
    console.log(response)
};
```

Which outputed:

```
INFO[0000] 301                                           source=console
INFO[0000] {"Date":"Mon, 03 Oct 2022 12:11:17 GMT","Content-Length":"6","Content-Type":"text/plain; charset=utf-8"}  source=console
INFO[0000] Hello!                                        source=console
INFO[0000] {"remote_ip":"127.0.0.1","remote_port":8181,"url":"http://localhost:8181/","status":301,"status_text":"301 Moved Permanently","proto":"HTTP/1.1","headers":{"Date":"Mon, 03 Oct 2022 12:11:17 GMT","Content-Length":"6","Content-Type":"text/plain; charset=utf-8"},"cookies":{},"body":"Hello!","timings":{"duration":0.340825,"blocked":1.765503,"looking_up":0,"connecting":1.627048,"tls_handshaking":0,"sending":0.055442,"waiting":0.20637,"receiving":0.079013},"tls_version":"","tls_cipher_suite":"","ocsp":{"produced_at":0,"this_update":0,"next_update":0,"revoked_at":0,"revocation_reason":"","status":""},"error":"","error_code":0,"request":{"method":"GET","url":"http://localhost:8181/","headers":{"User-Agent":["k6/0.40.0 (https://k6.io/)"]},"body":"","cookies":{}}}  source=console
```

Beatified JSON body:

```json
{
  "remote_ip": "127.0.0.1",
  "remote_port": 8181,
  "url": "http://localhost:8181/",
  "status": 301,
  "status_text": "301 Moved Permanently",
  "proto": "HTTP/1.1",
  "headers": {
    "Date": "Mon, 03 Oct 2022 12:11:17 GMT",
    "Content-Length": "6",
    "Content-Type": "text/plain; charset=utf-8"
  },
  "cookies": {},
  "body": "Hello!",
  "timings": {
    "duration": 0.340825,
    "blocked": 1.765503,
    "looking_up": 0,
    "connecting": 1.627048,
    "tls_handshaking": 0,
    "sending": 0.055442,
    "waiting": 0.20637,
    "receiving": 0.079013
  },
  "tls_version": "",
  "tls_cipher_suite": "",
  "ocsp": {
    "produced_at": 0,
    "this_update": 0,
    "next_update": 0,
    "revoked_at": 0,
    "revocation_reason": "",
    "status": ""
  },
  "error": "",
  "error_code": 0,
  "request": {
    "method": "GET",
    "url": "http://localhost:8181/",
    "headers": {
      "User-Agent": [
        "k6/0.40.0 (https://k6.io/)"
      ]
    },
    "body": "",
    "cookies": {}
  }
}
```

As you can see, response fields `body,` `status`, `headers` and `tls_properties`are not empty.

My k6 version:

```
k6 v0.40.0 (v0.40.0, go1.19.1, linux/amd64)
```

Can you provide more information on bug reproducibility?

**Comment by na--** [comment]
Closing this, since it's a duplicate of this issue: https://github.com/grafana/k6/issues/2474

The TLDR version is that the bug was due to Go 1.18 stdlib's incorrect handling of 30x responses and is fixed in Go 1.19. 

@atlekbai, the reason you don't see the bug is that your k6 v0.40.0 release was compiled with `go1.19.1`. The official k6 v0.40.0 packages (the [github release](https://github.com/grafana/k6/releases/tag/v0.40.0) binaries, the .msi, .rpm, .deb packages on [dl.k6.io](https://dl.k6.io/), the [0.40.0 docker image](https://hub.docker.com/layers/grafana/k6/0.40.0/images/sha256-239d51c6bebf02edc15ede33b122aec6d08e84f855e0226847830e9283c3bb66?context=explore)) were compiled with Go 1.18, but some external packages (e.g. the homebrew and AUR packages) were compiled with Go 1.19 and don't have the problem.

It will be fully fixed in k6 v0.41.0, but since https://github.com/grafana/k6/pull/2696 was merged, in the meantime you can use the `grafana/k6:master` docker image before that or build your own.

**Comment by mstoykov** [comment]
> bug was due to Go 1.18 stdlib's 

A small addition: the [upstream bug](https://github.com/golang/go/issues/49281) was reported against 1.17 *and* from a quick git archaeology - the responsible [line](https://github.com/golang/go/blame/88a06f40dfcdc4d37346be169f2b1b9070f38bb3/src/net/http/client.go#L645) has been around for 7 years at this point.
